src/sentry/api/endpoints/project_details.py

```diff
@@ -501,6 +501,14 @@ def get(self, request: Request, project) -> Response:
         if "hasAlertIntegration" in expand:
             data["hasAlertIntegrationInstalled"] = has_alert_integration(project)
 
+        is_am2_basic = features.has(
+            "organizations:dynamic-sampling-basic", project.organization, actor=request.user
+        )
+        # ToDo(ahmed): We need another check for developer AM2 plan as the AM2 Basic plan is only for Team and above
+        #  so we have no way of knowing if the a developer is on AM2
+        if is_am2_basic:
+            del data["dynamicSampling"]
```

Comment:
This will raise a KeyError if the key is not in the dictionary.

You can try more safe approach:
```
data.pop('dynamicSampling', None)
```

Reply:
Agreed!